app/components/post_draft/quick_actions/input_quick_action/index.tsx

```diff
@@ -35,27 +43,65 @@ export default function InputQuickAction({
     disabled,
     inputType,
     updateValue,
+    customEmojis = [],
     focus,
 }: Props) {
     const theme = useTheme();
+    const intl = useIntl();
     const onPress = useCallback(() => {
-        updateValue((v) => {
-            if (inputType === 'at') {
-                return `${v}@`;
-            }
-            return '/';
-        });
-        focus();
+        if (inputType === 'at') {
+            updateValue((v) => `${v}@`);
+            focus();
+        } else if (inputType === 'slash') {
+            updateValue((v) => `/${v}`);
+            focus();
+        } else {
+            openEmojiPicker();
+        }
     }, [inputType]);
 
-    const actionTestID = disabled ?
-        `${testID}.disabled` :
-        testID;
+    const handleEmojiClick = useCallback((emoji: string) => {
+        let emojiDraft: string;
+
+        const emojiData = getEmojiByName(emoji, customEmojis);
+
+        if (emojiData?.image && emojiData.category !== 'custom') {
+            const codeArray: string[] = emojiData.image.split('-');
+            const code = codeArray.reduce((acc, c) => {
+                return acc + String.fromCodePoint(parseInt(c, 16));
+            }, '');
+            emojiDraft = code;
+        } else {
+            emojiDraft = `:${emoji}: `;
+        }
```

Comment:
A great part of this logic probably comes from `components/autocomplete/emoji_suggestion/emoji_suggestion.tsx`. Any chance we can extract it to a common function in the utils folder?

Reply:
yeah, I feel too that it can be refactored, will do it